dimagi/commcare-hq
# Partition update_calculated_properties celery task

## Product Description

None
## Technical Summary

https://dimagi-dev.atlassian.net/browse/SAAS-12621

The `update_calculated_properties` celery task constantly fails because it takes too long to run. You can see in this [datadog graph](https://app.datadoghq.com/dashboard/bdu-k5b-bz5/celery-overview?fullscreen_end_ts=1633633019752&fullscreen_paused=false&fullscreen_section=overview&fullscreen_start_ts=1594148219752&fullscreen_widget=7941863544971934&tpl_var_celery_task=corehq.apps.reports.tasks.update_calculated_properties&from_ts=1594142752568&to_ts=1633627552568&live=true) that the task length has been increasing, and successful tasks are rare these days. 

The idea here is to use the power of multiple celery workers to split up this long running task into smaller, shorter child tasks. The partition boundaries are defined by domain names. 

## Feature Flag


## Safety Assurance

### Safety story

This task is already consistently failing, so it's only up from here. Changing the cache timeout for `all_domain_stats` will have an impact on how often the `commcare.mobile_workers.count` metric value is updated, but a day-to-day change of this metric is enough granularity for our needs. 

### Automated test coverage


No automated tests.
### QA Plan


Was not planning on running through QA. Staging does not have issues with this task currently, so the simplest check is to put this on staging and ensure the task is _still_ successful. We will only be able to tell if this fixes issues on production once it is actually rolled out to production.

### Migrations

- [x] The migrations in this code can be safely applied first independently of the code



### Rollback instructions




### Labels & Review

## Final self-reflection
- [x] My safety story above is convincing and gives me confidence that this PR will not introduce a regression
